FairSoftConfig.cmake

```diff
@@ -79,3 +79,79 @@ if(APPLE)
   string(STRIP "${macos_sdk_path}" macos_sdk_path)
   set(CMAKE_OSX_SYSROOT "${macos_sdk_path}" CACHE FILEPATH "macOS SDK" FORCE)
 endif()
+
+if(APPLE)
+  execute_process(COMMAND brew --prefix OUTPUT_VARIABLE brew_prefix)
+endif()
+
+#
+# LZMA
+#
+# On macOS LZMA isn't installed in some standard system path
+# so we need to take it from brew. The xz package which contains
+# the needed headers and libraries is installed with the setup script
+# from <source-dir>/legacy/setup-macos.sh.
+# On Linux the packages are in the default system paths and are found
+# automatically by Boost.
+#
+if(NOT ${brew_prefix} MATCHES "/usr/local")
+  execute_process(COMMAND brew list xz RESULT_VARIABLE lzma_installed OUTPUT_VARIABLE _tmp)
+  if (NOT lzma_installed)
+    message(STATUS "LZMA installation found. Boost iostream will be build with lzma compression")
+    execute_process(COMMAND brew --prefix xz OUTPUT_VARIABLE lzma_prefix)
+    string(STRIP "${lzma_prefix}" lzma_prefix)
+    set(LZMA_ROOT "${lzma_prefix}" CACHE FILEPATH "LZMA prefix" FORCE)
+    execute_process(COMMAND ${LZMA_ROOT}/bin/xz --version
+                    OUTPUT_VARIABLE lzma_version
+                   )
+    string(REGEX REPLACE ".*([0-9]\\.[0-9]\\.[0-9])+.*" "\\1" lzma_version1 ${lzma_version})
+    set(LZMA_EXTERNAL "using lzma   :" CACHE STRING "LZMA exetrnal usage for Boost" FORCE)
+    set(LZMA_VERSION "${lzma_version1}" CACHE STRING "LZMA version info for Boost" FORCE)
+    set(LZMA_STRING ": <include>${LZMA_ROOT}/include <search>${LZMA_ROOT}/lib ;" CACHE STRING "LZMA install info for Boost" FORCE)
+  else()
+    message(WARNING "LZMA installation not found. Boost iostream will be build without lzma compression")
+    set(LZMA_EXTERNAL "" CACHE STRING "LZMA exetrnal usage for Boost" FORCE)
+    set(LZMA_VERSION "" CACHE STRING "LZMA version info for Boost" FORCE)
+    set(LZMA_STRING "" CACHE STRING "LZMA install info for Boost" FORCE)
+  endif()
+else()
+  set(LZMA_EXTERNAL "" CACHE STRING "LZMA exetrnal usage for Boost" FORCE)
+  set(LZMA_VERSION "" CACHE STRING "LZMA version info for Boost" FORCE)
+  set(LZMA_STRING "" CACHE STRING "LZMA install info for Boost" FORCE)
+endif()
```

Comment:
It is not guaranteed that the user will use `-C FairSoftConfig.cmake`.

The dependency on `brew` should stay in this file to detect the lzma/zstd prefix and set the standard CMake prefix hints (`LIBLZMA_ROOT` and `LIBZSTD_ROOT`). The rest of the logic (version extraction) should go to `cmake/legacy.cmake` (conditionally on whether the prefix was passed in).

The msg output is not really correct. The script checks, if there is a brew installation of lzma and if not, it prints "Boost iostream will be build without lzma compression". How do you know this at this point? There can be other lzma installations present in the system?

Reply:
@dennisklein,

I put everything to cmake/legacy.cmake, please check the new MR #528. For LZMA there is a standard find module. For ZSTD I add one. CMake will try to find both packages independent of the system but will add the necessary information to the boost build system in case of macosx. I only tested the logic so far on my Apple silicon but will test it now also for my Intel based Mac and Debian 10.